docroot/modules/humsci/hs_migrate/hs_migrate.install

```diff
@@ -73,3 +73,76 @@ function hs_migrate_update_8006() {
     }
   }
 }
+
+/**
+ * Remove unused importers.
+ */
+function hs_migrate_update_8007() {
+  $unused_importers = [
+    'hs_d7_gallery_nodes',
+    'hs_d7_gallery_paragraphs',
+    'hs_d7_people',
+    'hs_d7_publications',
+    'hs_news_rss',
+    'custm_cesta_publications',
+    'custm_dlcl_research',
+    'custm_dlcl_research_unit_group',
+    'custm_em1060_manuscript',
+    'custm_facultyaffairs_users',
+    'custm_hs_d7_gc_publication_component',
+    'custm_hshr_client',
+    'custm_sgs_d7_degrees_offered',
+    'custm_sgs_d7_fund_opp',
+    'custm_sgs_d7_testimonials',
+    'custm_stats_tech_reports',
+    'custm_hs_wmlcp_users',
+    'hs_d7_news'
+  ];
+
+  // Get site name to prevent removing hs_d7_news from MRC.
+  $site_name = \Drupal::config('system.site')->get('name');
+  $is_mrc = str_contains($site_name, 'Mathematics Research Center');
+
+  $schema = \Drupal::database()->schema();
+
+  foreach ($unused_importers as $importer) {
+
+    if ($is_mrc && ($importer === 'hs_d7_news')) {
+      continue;
+    }
+
+    // Verify migration entity exists.
+    $migration = \Drupal::entityTypeManager()
+      ->getStorage('migration')
+      ->load($importer);
+
+    if(is_null($migration)){
+      continue;
+    }
+
+    // Remove migration importer configuration.
+    $migration->delete();
+    $message = t('Deleted @config', [
+      '@config' => 'migrate_plus.migration.' . $importer,
+    ]);
+    \Drupal::logger('hs_migrate')->notice($message);
```

Comment:
If you call a service multiple times in a loop, I prefer to set a variable outside the loop and use that. You did that for the schema service. ie
```
$logger = \Drupal::service('hs_migrate');
for ($foo as $bar) {
  $logger->notice('message');
}
```